test/alternator/conftest.py

```diff
@@ -76,110 +70,121 @@ def pytest_collection_modifyitems(config, items):
+                        pass
+        except NoHostAvailable:
+            # CQL is not available, so we can't find a valid role.
+            pass
+        # If we couldn't find a valid role, let's hope that
+        # alternator-enforce-authorization is not enabled so anything will work
+        return ('unknown_user', 'unknown_secret')
+
+    return _get_valid_alternator_role
+
+
+@pytest.fixture(scope="module")
+def local_url(request):
+    # Even though we connect to the local installation, Boto3 still
+    # requires us to specify dummy region and credential parameters,
+    # otherwise the user is forced to properly configure ~/.aws even
+    # for local runs.
+    if request.config.getoption("url") is not None:
+        return request.config.getoption("url")
+
+    if host := request.getfixturevalue("host"):
+        # this argument needed for compatibility with PythonTestSuite without modifying the previous behavior
+        return f"http://{host}:8000"
+
+    if request.config.getoption("https"):
+        return "https://localhost:8043"
+
+    return "http://localhost:8000"
+
 
 # "dynamodb" fixture: set up client object for communicating with the DynamoDB
 # API. Currently this chooses either Amazon's DynamoDB in the default region
 # or a local Alternator installation on http://localhost:8080 - depending on the
 # existence of the "--aws" option. In the future we should provide options
 # for choosing other Amazon regions or local installations.
 # We use scope="session" so that all tests will reuse the same client object.
-@pytest.fixture(scope="session")
-def dynamodb(request):
+@pytest.fixture(scope="module")
+def dynamodb(request, get_valid_alternator_role):
     # Disable boto3's client-side validation of parameters. This validation
     # only makes it impossible for us to test various error conditions,
     # because boto3 checks them before we can get the server to check them.
     boto_config = botocore.client.Config(parameter_validation=False)
     if request.config.getoption('aws'):
         res = boto3.resource('dynamodb', config=boto_config)
     else:
-        # Even though we connect to the local installation, Boto3 still
-        # requires us to specify dummy region and credential parameters,
-        # otherwise the user is forced to properly configure ~/.aws even
-        # for local runs.
-        if request.config.getoption('url') != None:
-            local_url = request.config.getoption('url')
-        elif request.config.getoption('host') is not None:
-            # this argument needed for compatibility with PythonTestSuite without modifying the previous behavior
-            local_url = f"http://{request.config.getoption('host')}:8000"
-        else:
-            local_url = 'https://localhost:8043' if request.config.getoption('https') else 'http://localhost:8000'
-        # Disable verifying in order to be able to use self-signed TLS certificates
+        endpoint_url = request.getfixturevalue("local_url")
```

Comment:
Why not to use fixture local_url?
I know with this way, we are triggering it only when it's needed, but it's pretty simple and lightweight.
Moreover, wouldn't it be better to return a function that takes a parameter 'dynamodb' or 'dynamostreams', so you will remove duplication here and in the dynamodbstreams()?

Reply:
Because if `--aws` argument is provided I don't want to execute `local_url` fixture at all.

Anyway, @nyh asked to remove this de-duplication.

Reply:
Removed `local_url` fixture from this PR.